llvm/lib/Target/AIE/AIEInterBlockScheduling.cpp

```diff
@@ -169,7 +173,8 @@ bool InterBlockScheduling::leaveBlock() {
     // If we are very unlucky, we may step both the latency margin and
     // the resource margin to the max. Any more indicates failure to converge,
     // and we abort to prevent an infinite loop.
-    if (BS.FixPoint.NumIters > 2 * HR->getConflictHorizon()) {
+    if (BS.FixPoint.NumIters >
+        2 * HR->getConflictHorizon() + MaxExpensiveIterations) {
```

Comment:
nit: I think we first do MaxExpensiveIterations, then fall back to the global safety margins. Reverse the two terms here?